pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -193,6 +193,14 @@ func (binder *PersistentVolumeClaimBinder) deleteClaim(obj interface{}) {
 func syncVolume(volumeIndex *persistentVolumeOrderedIndex, binderClient binderClient, volume *api.PersistentVolume) (err error) {
 	glog.V(5).Infof("Synchronizing PersistentVolume[%s], current phase: %s\n", volume.Name, volume.Status.Phase)
 
+	// The PV may have been modified by parallel call to syncVolume, load
+	// the current version.
+	newPv, err := binderClient.GetPersistentVolume(volume.Name)
+	if err != nil {
+		return fmt.Errorf("Cannot reload volume %s: %v", volume.Name, err)
+	}
+	volume = newPv
```

Comment:
We use the volumeIndex custom Store impl because we needed fancy indexing and sorting.  The watch still has its own store.

We would need to add the PV to both.  

Ideally, we allow the Controller framework to take a store argument it uses for watch data.  It does not currently, so I couldn't pass our volumeIndex to it.  This controller doesn't need a 2nd store. We need to be able to specify the one we want to use in the watch.